### PR TITLE
Add a newline to pem blobs if one is not present

### LINKED ISSUE #504: Loading Kubeconfig fails if it contains pems with no trailing newline
X509::from_pem fails if the kubeconfig has been constructed so that the pkey or certs are missing a trailing newline.  By the spec, they're *supposed* to have a trailing newline after the `----- END RSA PRIVATE KEY -----`  but hey, stuff happens.  

Kubectl tolerates this scenario, so we should ensure that kube meets those expectations.

### PR DESCRIPTION
**Pull request by rrichardson**
/fixes #504

I'm not sure where is the best place to put this routine. 
The function I chose isn't technically limited to loading PEM files, but it's currently not used for anything else. 
It's the only place I found where we'd only have to invoke the routine once. 

The only actual change is in `src/config/utils.rs`  the rest are formatting changes.

### CODE REVIEW

**Comment by kazk** [comment]
> The only actual change is in `src/config/utils.rs` the rest are formatting changes.

@clux Looks like rustfmt job failed for some reason. https://github.com/clux/kube-rs/runs/2423729395

**Comment by rrichardson** [comment]
@clux - Those were formatted with `cargo +nightly fmt`

**Comment by kazk** [comment]
@rrichardson 

Removing other files from the PR like the following should be enough?
```bash
git reset HEAD~
git add kube/src/config/utils.rs
git restore . # restore other files
git commit
git push --force-with-lease
```

Other files should be formatted by GitHub Actions after merging. The rustfmt job failed last time for unknown reason.

File: kube/src/config/utils.rs [hunk 1]
```diff
@@ -12,13 +12,21 @@ pub fn default_kube_path() -> Option<PathBuf> {
...
+    let mut blob = match (data, file) {
         (Some(d), _) => base64::decode(&d)
             .map_err(ConfigError::Base64Decode)
             .map_err(Error::Kubeconfig),
         (_, Some(f)) => read_file(f),
         _ => Err(ConfigError::NoBase64FileOrData.into()),
+    };
+    //Ensure there is a trailing newline in the blob
+    //Don't bother if the blob is empty
+    if let Ok(buf) = &mut blob {
```
[the remark this answers is not in the document]
Comment (rrichardson) [reply kube/src/config/utils.rs]: I used your suggestion, but went with ye olde question mark instead.  I think it more clearly indicates the intent. 
https://github.com/clux/kube-rs/blob/2ee23254e3a2305cf85e2a63cc5efdca93710663/kube/src/config/utils.rs#L14-L28

**Review by clux — APPROVED** [review]

**Review by kazk — APPROVED** [review]

**Comment by kazk** [comment]
> Other files should be formatted by GitHub Actions after merging.

I thought I had seen GitHub Action formatting code and pushing commit when necessary, but I guess I misremembered (maybe it was from other repo).

**Comment by clux** [comment]
If it's possible to do that, then that's great. Usually I have been doing a quick `make fmt && git ci -m fmt` after commits (since github stuff doesn't run on forks).

Turns out the stray formatting issues in there was my fault - had merged some prs and forgotten to do it.

**Comment by clux** [comment]
Released in 0.53.0.

**Pull request merged**